src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportControlCheck.java

```diff
@@ -559,7 +559,7 @@ else if (currentImportControl != null) {
 
     @Override
     public Set<String> getExternalResourceLocations() {
-        return Collections.singleton(file.toString());
+        return Collections.singleton(file.toASCIIString());
```

Comment:
as pitest failures suggested, we need test coverage.

Please create junit test method that creates temporary file in temp folder of OS with name to have unicode character.
We need to copy Input file (you can find a lot examples in repo) to this folder and execute Check validation on this Input.

Reply:
@reftel , let us know if you need a bit more guidance or explanation.

Reply:
```
[INFO] --- forbiddenapis:3.4:check (forbiddenapis-main) @ checkstyle ---
[WARNING] Parameter 'localRepository' is deprecated core expression; Avoid use of ArtifactRepository type. If you need access to local repository, switch to '${repositorySystemSession}' expression and get LRM from it instead.
[INFO] Scanning for classes to check...
[INFO] Reading bundled API signatures: jdk-unsafe-11
[INFO] Reading bundled API signatures: jdk-deprecated-11
[INFO] Reading bundled API signatures: jdk-system-out
[INFO] Reading bundled API signatures: jdk-non-portable
[INFO] Reading API signatures: /Users/[username]/src/checkstyle/checkstyle/config/signatures.txt
[INFO] Loading classes to check...
[INFO] Scanning classes for violations...
[ERROR] Forbidden method invocation: java.net.URI#toString() [This method can return garbage for non-ascii data, please use URI#toASCIIString()]
[ERROR]   in com.puppycrawl.tools.checkstyle.ConfigurationLoader (ConfigurationLoader.java:265)
[ERROR] Scanned 840 class file(s) for forbidden API invocations (in 1.65s), 1 error(s).
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  12:16 min
[INFO] Finished at: 2023-04-21T23:07:22+02:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal de.thetaphi:forbiddenapis:3.4:check (forbiddenapis-main) on project checkstyle: Check for forbidden API calls failed, see log. -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
```